include/pybind11/detail/common.h

```diff
@@ -1037,7 +1037,7 @@ template <typename... Args>
         || (defined(__clang__) && __clang_major__ == 5)
 static constexpr detail::overload_cast_impl<Args...> overload_cast = {};
 #    else
-static constexpr detail::overload_cast_impl<Args...> overload_cast;
+static constexpr detail::overload_cast_impl<Args...> overload_cast{};
```

Comment:
Uhm ... wait, do we actually need this ugly `#ifdef`?

Reply:
You mean maybe the new version works on all compilers?

Reply:
Also why weren't we seeing the errors with newer MSVC? We should be testing with 2019 & 2022 (maybe not 2017 anywhere anymore).

Reply:
> You mean maybe the new version works on all compilers?

Yes, I tried it out, CI is green: https://github.com/pybind/pybind11/pull/4190

Whatever made the `#ifdef` necessary in the past, it doesn't seem to be an issue anymore with the compilers we support today.

Reply:
> Uhm ... wait, do we actually need this ugly #ifdef?

Actually if the `{}` is present, this `#ifdef` can be removed. I will cover that in next commit and run checks.

> Also why weren't we seeing the errors with newer MSVC? 

My bad, I have got lost in the logs of the project and actual MSVC build version is `1924`. Maybe this version is not covered? Also I see that `-DCMAKE_CXX_STANDARD=17/20` is set for Windows builds, is `C++14` covered as well?

Reply:
I bet no one ever tried it with `{}`, but only with `= {}`.

I'm hoping to get a patch release out in ~1 week or so. Still have a few items on #4125 left.

Reply:
@henryiii Thanks for the answer, can't wait to finally bump version in OpenVINO!

Reply:
@henryiii As fix version 2.10.1 is going to be released, will it include this PR and everything from the master branch up to this point? I am not accustom to pybind's release practices, I am really sorry for such questions.

(I do not want to spam in #4125 comments, hope you don't mind:))

Reply:
It should. I think the two breaking changes might go in before it releases, which means we'll branch, but so far master is 2.10.1. I'm going to finish investigation into the CMake caching issue in the next couple of days, then make a release.

Reply:
Missed my target, but 2.10.1 is out now.